### PR TITLE
Create documentation

### PR DESCRIPTION
**Pull request by abelsiqueira**

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# NLPModelsJuMP.jl documentation
+
+This package defines a
+[`NLPModels`](https://github.com/JuliaSmoothOptimizers/NLPModels.jl) model using
+[`MathProgBase`](https://github.com/JuliaOpt/MathProgBase.jl) and
+[`JuMP.jl`](https://github.com/JuliaOpt/JuMP.jl).
+This documentation is specific for this model. Please refer to the [documentation of
+NLPModels](http://juliasmoothoptimizers.github.io/NLPModels.jl/stable/) if in doubt.
```
Comment (dpo) [inline docs/src/index.md]: "documentation of NLPModels" -> "NLPModels documentation"

File: docs/src/tutorial.md [hunk 2]
```diff
@@ -0,0 +1,225 @@
+# Tutorial
+
+NLPModelsJuMP is a combination of NLPModels and JuMP, as the name imply.
```
Comment (dpo) [inline docs/src/tutorial.md]: "imply" -> "implies"

File: docs/src/tutorial.md [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+Sometimes it may be required to refer to the specific documentation, as we'll present
+here the result of combining both.
```
Comment (dpo) [inline docs/src/tutorial.md]: I'm not sure I understand what you're saying here. Does "both" refer to NLPModels and JuMP models?

File: docs/src/tutorial.md [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+```
+
+## MathProgNLPModel
+
+```@docs
+MathProgNLPModel
+```
+
+`MathProgNLPModel` is a simple yet efficient model. It uses JuMP to define the problem,
+which can then be accessed through the NLPModels API.
```
Comment (dpo) [inline docs/src/tutorial.md]: "which can then" -> "and can"

File: docs/src/tutorial.md [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+Using `ADNLPModel` is simpler, as it comes by default, but `ADNLPModel` doesn't handle
+sparse derivatives and `MathProgNLPModel` does.
```
Comment (dpo) [inline docs/src/tutorial.md]: How about "An advantage of `MathProgNLPModel` over simpler models such as `ADNLPModel`s is that they provide sparse derivatives."?

File: docs/src/tutorial.md [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+
+x0 = [-1.2; 1.0]
+model = Model() # No solver is required
+@variable(model, x[i=1:2], start=x0[i])
+@NLobjective(model, Min, (x[1] - 1)^2 + 100 * (x[2] - x[1]^2)^2)
+
+nlp = MathProgNLPModel(model)
+```
+
+This defines the model.
```
Comment (dpo) [inline docs/src/tutorial.md]: I don't think this is necessary.

File: docs/src/tutorial.md [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+Let's get the objective function value at ``x^0``, using only `nlp`.
+
+```@example jumpnlp
+fx = obj(nlp, nlp.meta.x0)
+println("fx = $fx")
+```
+
+Done.
```
Comment (dpo) [inline docs/src/tutorial.md]: I don't think this is necessary.

File: docs/src/tutorial.md [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+
+adnlp = ADNLPModel(f, x0)
+x, fx, ngx, optimal, iter = steepest(adnlp)
+```
+
+### OptimizationProblems
+
+The package
+[OptimizationProblems](https://github.com/JuliaSmoothOptimizers/OptimizationProblems.jl)
+provides a reasonable amount of problems defined in JuMP format, which can be converted
```
Comment (dpo) [inline docs/src/tutorial.md]: "reasonable amount" -> "collection"

File: docs/src/tutorial.md [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+
+nlp = MathProgNLPModel(woods())
+x, fx, ngx, optimal, iter = steepest(nlp)
+println("fx = $fx")
+println("ngx = $ngx")
+println("optimal = $optimal")
+println("iter = $iter")
+```
+
+Constrained problem can also be converted.
```
Comment (dpo) [inline docs/src/tutorial.md]: problems

File: docs/src/tutorial.md [hunk 10]
```diff
@@ -0,0 +1,225 @@
...
+```@docs
+MathProgNLSModel
+```
+
+`MathProgNLSModel` is a model for nonlinear least squares using JuMP.
+To use it, we define a JuMP model without the objective, and use `NLexpression`s to
+define the residual function.
+For instance, the Rosenbrock function in nonlinear least squares format is
+```math
+F(x) = \begin{bmatrix} x_1 - 1\\ 10(x_2 - x_1^2) \end{bmatrix},
```
Comment (dpo) [inline docs/src/tutorial.md]: The Rosenbrock function is really 1/2 * ||F(x)||.

File: docs/src/tutorial.md (hunk 3 above)
Comment (abelsiqueira) [reply docs/src/tutorial.md]: Yes, but I'm thinking now "we'll present here only the documentation specific to NLPModelsJuMP".

Comment (dpo) [reply docs/src/tutorial.md]: That works. Thank you!

File: docs/src/tutorial.md (hunk 10 above)
Comment (abelsiqueira) [reply docs/src/tutorial.md]: I've tried to write in a clearer way.

Comment (dpo) [reply docs/src/tutorial.md]: That's great, thank you.

**Comment by abelsiqueira** [comment]
Updated

**Comment by dpo** [comment]
Thank you!

**Pull request merged**